### PR TITLE
add software-version test to ensure the string is non empty.

### PR DESCRIPTION
**Pull request by yannbua**
the software-version value is useful to perform the inventory of network devices.
network vendors may set it in native models or in components and may miss setting it for the main system on this xpath.

### CODE REVIEW

**Comment by Swetha-haridasula** [comment]
@yannbua please also add the Cannocal OC for Software-version to the README

File: feature/system/system_base_test/tests/system_software_version/system_software_version_test.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import (
+	"testing"
+
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+)
+
+// TestSoftwareVersion verifies that the software version state path can be read and is not empty.
```
Comment (bstoll) [inline feature/system/system_base_test/tests/system_software_version/system_software_version_test.go:27]: The function comment should appear directly above the `func TestSoftwareVersion` definition (no newlines between) for it to be picked up by godoc/IDEs.

File: feature/system/system_base_test/tests/system_software_version/system_software_version_test.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+// telemetry_path:/system/state/software-version
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+func TestSoftwareVersion(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	state := gnmi.OC().System().SoftwareVersion()
+	t.Run("Test Software Version", func(t *testing.T) {
```
Comment (bstoll) [inline feature/system/system_base_test/tests/system_software_version/system_software_version_test.go:37]: If there is only 1 thing being tested, there isn't a value defining a subtest - we can just have the lines inside t.Run() be pulled out.  If additional things get added in the future then subtests (or multiple test functions) would make sense to add at that time.

File: feature/system/system_base_test/tests/system_software_version/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+Each test will require the DUT configured with a basic service configuration that
+should be provided as part of the basic configuration. This setup should also include
+any security setup for connecting to the services.
+
+### Tests
+
+1. Configure DUT with service configurations for all required services
+2. The test will verify if the software-version state path can be read and is non-empty.
+
```
Comment (Swetha-haridasula) [inline feature/system/system_base_test/tests/system_software_version/README.md:17]: Please add JSON Canonical OC

File: feature/system/system_base_test/tests/system_software_version/system_software_version_test.go (hunk 2 above)
Comment (yannbua) [reply feature/system/system_base_test/tests/system_software_version/system_software_version_test.go]: Thanks. Done.

File: feature/system/system_base_test/tests/system_software_version/system_software_version_test.go (hunk 1 above)
Comment (yannbua) [reply feature/system/system_base_test/tests/system_software_version/system_software_version_test.go]: done.
We have few more tests in system_base_test that follow this pattern.
If I find time, I will update them too.

**Comment by yannbua** [comment]
@Swetha-haridasula or @bstoll, PTAL. Comments are addressed. Checks are passing again with the changes.

**Comment by yannbua** [comment]
@bstoll or @Swetha-haridasula  could you merge the PR if everything looks fine for you ? I will sync once more my branch.

File: feature/system/system_base_test/tests/system_software_version/system_software_version_test.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// TestSoftwareVersion verifies that the software version state path can be read and is not empty.
+// telemetry_path:/system/state/software-version
+func TestSoftwareVersion(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	state := gnmi.OC().System().SoftwareVersion()
+	stateGot := gnmi.Get(t, dut, state.State())
```
Comment (singhavnish2516) [inline feature/system/system_base_test/tests/system_software_version/system_software_version_test.go:36]: Add this in the if statement please? Also, `got` should be enough.

File: feature/system/system_base_test/tests/system_software_version/system_software_version_test.go [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// TestSoftwareVersion verifies that the software version state path can be read and is not empty.
+// telemetry_path:/system/state/software-version
+func TestSoftwareVersion(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	state := gnmi.OC().System().SoftwareVersion()
```
Comment (singhavnish2516) [inline feature/system/system_base_test/tests/system_software_version/system_software_version_test.go:35]: Misleading name. Please see if we can/should move this to the function call itself. If that does not look a good idea, please use a name like `swVersion`?

**Review by singhavnish2516 — CHANGES_REQUESTED** [review]

File: feature/system/system_base_test/tests/system_software_version/system_software_version_test.go (hunk 5 above)
Comment (yannbua) [reply feature/system/system_base_test/tests/system_software_version/system_software_version_test.go]: I initially copied the hostname test as a basis as I assumed it was a good boilerplate to start from.
gnmi.OC() is now moved to the gnmi.Get() call.

File: feature/system/system_base_test/tests/system_software_version/system_software_version_test.go (hunk 4 above)
Comment (yannbua) [reply feature/system/system_base_test/tests/system_software_version/system_software_version_test.go]: done.

**Comment by yannbua** [comment]
> @yannbua please help address comments to get it merged, thanks

done. I was on vacation hence the latency. Thanks for the suggested improvements.

**Review by singhavnish2516 — APPROVED** [review]
